File: app/src/main/java/com/iffy/mianshi/basic/fanxing/rx/Observable.java
package com.iffy.mianshi.basic.fanxing.rx;

/**
 * author : iffy
 * time   : 2020/03/12
 */
//将泛型声明在类上
public interface Observable<T> {
    T call();

    <U> Observable<U> map(Func1<T, U> func1);

    //Observable<T> doOnNext(Action<T> action);

    void doOnNext(Action<T> action);
}

File: app/src/main/java/com/iffy/mianshi/pattern/II观察者模式/test/Observable.kt
package com.iffy.mianshi.pattern.II观察者模式.test

/**
 *     author : iffy
 *     time   : 2020/03/12
 */
interface Observable {
    fun regist(observer: Observer)
    fun unregist(observer: Observer)
    fun notifyThem(s:String)
}